sci-libs/magma/magma-2.6.2-r1.ebuild

```
@@ -0,0 +1,195 @@
+	amdgpu_gfx1100 amdgpu_gfx1101 amdgpu_gfx1102 amdgpu_gfx1103
+"
+#IUSE="doc openblas test ${IUSE_AMDGPU}"
+IUSE="ampere doc cuda hip kepler maxwell openblas pascal test turing video_cards_nvidia volta ${IUSE_AMDGPU}"
+#kepler maxwell pascal volta turing ampere
+
+# TODO: do not enforce openblas
+RDEPEND="
+	openblas? ( sci-libs/openblas )
+	!openblas? (
+		virtual/blas
+		virtual/lapack
+	)
+"
+DEPEND="
+	${RDEPEND}
+	${PYTHON_DEPS}
+	cuda? ( dev-util/nvidia-cuda-toolkit )
+	hip? ( dev-util/hip )
+"
+#	dev-util/hip
+BDEPEND="
+	virtual/pkgconfig
+	doc? ( >=app-doc/doxygen-1.8.14-r1[dot] )
+"
+
+REQUIRED_USE="
+	^^ ( cuda hip )
+	ampere? ( cuda )
+	cuda? ( video_cards_nvidia || ( ampere kepler maxwell pascal turing volta ) )
+	hip? ( || ( ${IUSE_AMDGPU/+/} ) )
+	kepler? ( cuda )
+	maxwell? ( cuda )
+	pascal? ( cuda )
+	turing? ( cuda )
+	volta? ( cuda )
+"
+RESTRICT="!test? ( test )"
+
+pkg_setup() {
+	[[ MERGE_TYPE != binary ]] && tc-check-openmp || die "Need OpenMP to compile ${P}"
+	fortran-2_pkg_setup
+	python-any-r1_pkg_setup
+}
+
+src_prepare() {
+#	set_cuda_gpu_target() {
+#		if use ampere; then
+#			GPU_TARGET=Ampere
+#		elif use kepler; then
+#			GPU_TARGET=Kepler
+#		elif use maxwell; then
+#			GPU_TARGET=Maxwell
+#		elif use pascal; then
+#			GPU_TARGET=Pascal
+#		elif use turing; then
+#			GPU_TARGET=Turing
+#		elif use volta; then
+#			GPU_TARGET=Volta
+#		fi
+#		export GPU_TARGET
+#	}
+
+#	set_amd_gpu_target() {
+#	}
+
+	use cuda && cuda_src_prepare
+
+	rm -r blas_fix || die
+	# distributed pc file not so useful so replace it
+	cat <<-EOF > ${PN}.pc
+		prefix=${EPREFIX}/usr
+		libdir=\${prefix}/$(get_libdir)
+		includedir=\${prefix}/include/${PN}
+		Name: ${PN}
+		Description: ${DESCRIPTION}
+		Version: ${PV}
+		URL: ${HOMEPAGE}
+		Libs: -L\${libdir} -lmagma
+		Libs.private: -lm -lpthread -ldl
+		Cflags: -I\${includedir}
+		Requires: $(usex openblas "openblas" "blas lapack")
+	EOF
+
+# not necessary, according to README, only needed for live ebuilds
```

I prefer regenerating things because we are gentoo and we build from source

---

So we won't be able to build for CUDA. It fails early in the make generate step.